scripts/source-maintenance.sh

```diff
@@ -84,27 +101,37 @@ done
 # an error code seen by a KeepGoing-aware command (or zero)
 SeenErrors=0
 
-
 if ! git diff --quiet; then
 	echo "There are unstaged changes. This script may modify sources."
 	echo "Stage changes to avoid permanent losses when things go bad."
 	exit 1
 fi
 
-# On squid-cache.org we have to use the python scripted md5sum
-HOST=`hostname`
-if test "$HOST" = "squid-cache.org" ; then
-	MD5="md5"
+for Checksum in md5sum md5 shasum sha1sum
+do
+    if "$Checksum" </dev/null >/dev/null 2>/dev/null ; then
+        break
+    fi
+done
+echo "detected checksum program $Checksum"
+
+if [ "x$ASTYLE" != "x" ] ; then
+    if ${ASTYLE} --version >/dev/null 2>/dev/null ; then
+        :
+    else
+        echo "ERROR: cannot run user-supplied astyle ${ASTYLE}"
+    fi
 else
-	MD5="md5sum"
+    for AttemptedBinary in astyle-${TargetAstyleVersion} astyle
+    do
+        if $AttemptedBinary --version >/dev/null 2>/dev/null ; then
+            ASTYLE=$AttemptedBinary
+            echo "detected astyle program ${ASTYLE}"
+            break
+        fi
+    done
 fi
 
-${ASTYLE} --version >/dev/null 2>/dev/null
-result=$?
-if test $result -gt 0 ; then
-	echo "ERROR: cannot run ${ASTYLE}"
```

Comment:
Please quit if running `${ASTYLE}` does not work. That is what the script used to do, and that is still the best behavior AFAICT. This principle applies to manually-specified astyle spelling and AttemptedBinary tests, but there are many ways to address this change request, and I do not insist on any specific one, but, ideally, we should not run the same astyle test twice.

Reply:
Ok